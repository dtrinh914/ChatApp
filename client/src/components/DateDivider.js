import React from 'react'
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';
import {makeStyles} from '@material-ui/styles';

const useStyle = makeStyles({
    root:{
        position: 'relative',
        width: '100%',
        padding: '20px 0',
    },
    text:{
        position:'absolute',
        zIndex: 80,
        top: 0,
        left: '50%',
        transform: 'translateX(-50%)',
        backgroundColor: '#fafafa',
        padding: '8px 12px',
        borderRadius: '25px'
    },
    divider:{
        width:'95%',
        margin: '0 auto',
    }
})

export default function DateDivider({date}) {
    const classes = useStyle();

    return (
        <div className={classes.root}>
            <Typography className={classes.text}>{date}</Typography>
            <Divider className={classes.divider} />
        </div>
    )
}
