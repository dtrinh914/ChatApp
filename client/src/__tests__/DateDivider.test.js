import React from 'react';
import DateDivider from '../components/DateDivider';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const testDate = "Wed January 22, 2020";

test('should match snapshot', ()=>{
    const {container} = render(<DateDivider date={testDate} />)
    expect(container).toMatchSnapshot();
});

test('should display the test date', ()=>{
    const {container} = render(<DateDivider date={testDate} />)
    expect(container).toHaveTextContent(testDate);
})