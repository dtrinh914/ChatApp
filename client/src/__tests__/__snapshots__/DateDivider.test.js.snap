// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
    data-testid="date-divider"
  >
    <p
      class="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
    >
      Wed January 22, 2020
    </p>
    <hr
      class="MuiDivider-root makeStyles-divider-3"
    />
  </div>
</div>
`;
